### PR TITLE
Similarity Computer plugin

### PR DESCRIPTION
**Pull request by seinecle**
Hello Eduardo,

Here is another plugin migrated to 0.9.1! Your explanations from the last migrations were so useful to get it done alone this time!

### CODE REVIEW

**Comment by eduramiba** [comment]
Awesome! I will get to this pull request soon 😃

File: modules/GephiSimilarityComputer/src/main/java/Controller/MyFileImporter.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package Controller;
```
Comment (eduramiba) [inline modules/GephiSimilarityComputer/src/main/java/Controller/MyFileImporter.java]: Can you refactor all your files to include java.net.clementlevallois or something before each package? This is automatically done by Netbeans when you rename the package.

File: modules/GephiSimilarityComputer/src/main/java/Controller/MyImporterBuilder.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
```
Comment (eduramiba) [inline modules/GephiSimilarityComputer/src/main/java/Controller/MyImporterBuilder.java]: Also please remove the empty template header from all files and make sure authorship is correct in each file.

File: modules/GephiSimilarityComputer/src/main/java/Gaze/CosineCalculation.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+
+        //this looks complicated but is simply the number of elements in the networks and all their combinations
+        numCalculations = (int)Math.pow(numNodes, 2) / 2;
+
+
+        //this is where the adjacency matrix for the final network is built
+        Controller.similarityMatrix = new FlexCompColMatrix(numNodes, numNodes);
+
+        System.out.println("size of the similarity matrix: " + numNodes + " x " + numNodes);
```
Comment (eduramiba) [inline modules/GephiSimilarityComputer/src/main/java/Gaze/CosineCalculation.java]: All System.out should be removed or changed to Logger info.

File: modules/GephiSimilarityComputer/pom.xml [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>3.4</version>
+            <type>jar</type>
+        </dependency>
+    </dependencies>
+    <repositories>
```
Comment (eduramiba) [inline modules/GephiSimilarityComputer/pom.xml]: Is this repository necessary?

File: modules/GephiSimilarityComputer/pom.xml [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>nbm-maven-plugin</artifactId>
+                <configuration>
+                    <licenseName>Apache v2</licenseName>
+                    <licenseFile>Apache v2.txt</licenseFile>
+                    <author>Clement Levalllois (@seinecle)</author>
+                    <sourceCodeUrl>$sourcecode_url</sourceCodeUrl>
```
Comment (eduramiba) [inline modules/GephiSimilarityComputer/pom.xml]: This is a bug in the plugin creator, please change the URL to https://github.com/seinecle/gephi-plugins/tree/similaritycomputer

File: modules/GephiSimilarityComputer/pom.xml [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                    <homePageUrl>http://www.clementlevallois.net</homePageUrl>
+                    <publicPackages>
+                        
+                    </publicPackages>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (eduramiba) [inline modules/GephiSimilarityComputer/pom.xml]: Is Java 8 actually necessary? Otherwise you can remome these properties or set them to Java 7

**Comment by eduramiba** [comment]
Hi Clement,
Were you able to check the suggested changes?

If not, I can do it sometime but not sure when I will have the time.

**Comment by seinecle** [comment]
I am on it, I actually take this opportunity to revamp it

**Comment by seinecle** [comment]
Ok here is my new pull request - made the changes you recommended, added new stuff, refactored a lot. Much cleaner.

**Comment by eduramiba** [comment]
Hi,
A quick look at the code still shows a few problems:

- The Wizard package is still not refactored to net.clementlevallois
- The sourcecode url in pom.xml is still missing
- There are calls to System.out, logger should be used instead
- The sonatype-snapshots is still in pom, and not used by any dependency
- There seems to be a lot of unnecessary Bundle.properties files in src/main/resources. It seems only Wizard and Controller folders should be 

Thanks!

**Comment by seinecle** [comment]
OK thanks for your the code review!! Doing it today.

**Comment by seinecle** [comment]
Updated according to your comments

**Comment by eduramiba** [comment]
Merged, thanks! Will be published shortly

**Comment by eduramiba** [comment]
Your plugin is now available :) https://gephi.org/plugins/#/plugin/similaritycomputer

**Comment by seinecle** [comment]
Fantastic - thx for the great work!

**Pull request merged**